nilearn/maskers/nifti_labels_masker.py

```diff
@@ -396,114 +374,113 @@ def _reporting(self):
-                    msg = (
-                        "A list of 4D subject images were provided to fit. "
-                        "Only first subject is shown in the report."
-                    )
-                    warnings.warn(msg, stacklevel=find_stack_level())
-                    self._report_content["warning_message"] = msg
-                display = plotting.plot_img(
-                    img,
-                    cut_coords=cut_coords,
-                    black_bg=False,
-                    cmap=self.cmap,
-                )
-                plt.close()
-                display.add_contours(labels_image, filled=False, linewidths=3)
+        img = self._reporting_data["img"]
 
-            # Otherwise, simply plot the ROI of the label image
-            # and give a warning to the user
-            else:
+        # compute the cut coordinates on the label image in case
+        # we have a functional image
+        cut_coords = plotting.find_xyz_cut_coords(
+            labels_image, activation_threshold=0.5
+        )
+
+        # If we have a func image to show in the report, use it
+        if img is not None:
+            if self._reporting_data["dim"] == 5:
                 msg = (
-                    "No image provided to fit in NiftiLabelsMasker. "
-                    "Plotting ROIs of label image on the "
-                    "MNI152Template for reporting."
+                    "A list of 4D subject images were provided to fit. "
+                    "Only first subject is shown in the report."
                 )
                 warnings.warn(msg, stacklevel=find_stack_level())
                 self._report_content["warning_message"] = msg
-                display = plotting.plot_roi(labels_image)
-                plt.close()
-
-            # If we have a mask, show its contours
-            if self._reporting_data["mask"] is not None:
-                display.add_contours(
-                    self._reporting_data["mask"],
-                    filled=False,
-                    colors="g",
-                    linewidths=3,
-                )
+            display = plotting.plot_img(
+                img,
+                cut_coords=cut_coords,
+                black_bg=False,
+                cmap=self.cmap,
+            )
+            plt.close()
+            display.add_contours(labels_image, filled=False, linewidths=3)
+
+        # Otherwise, simply plot the ROI of the label image
+        # and give a warning to the user
         else:
-            self._report_content["summary"] = None
-            display = None
+            msg = (
+                "No image provided to fit in NiftiLabelsMasker. "
+                "Plotting ROIs of label image on the "
+                "MNI152Template for reporting."
+            )
+            warnings.warn(msg, stacklevel=find_stack_level())
+            self._report_content["warning_message"] = msg
+            display = plotting.plot_roi(labels_image, cmap=self.lut_)
```

Comment:
```suggestion
            display = plotting.plot_roi(labels_image)
```

Reply:
passing colors from the look up table to plotting will be done in a follow up PR: needs to see if this can be done also for surface label masker